src/modules/keyserver/key-server.usecase.ts

```diff
@@ -1,46 +1,137 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { UserAttributes } from '../user/user.attributes';
-import { Keys } from './key-server.domain';
+import {
+  KeyServer,
+  KeyServerAttributes,
+  UserKeysEncryptVersions,
+} from './key-server.domain';
 import { SequelizeKeyServerRepository } from './key-server.repository';
 
+export class InvalidKeyServerException extends Error {
```

Comment:
Extend from a BadRequestException instead of from a generic Error, we coupled the code to nestjs as it itself abstracts these things.